### PR TITLE
[IMP] mrp: update bills of materials

### PR DESCRIPTION
**Pull request by brse-odoo**
This PR is a targeted revision for V14.0 only.

Performed complete rewrite for MRP document: bill_configuration.rst

-Added new sections to document (based off 15.0/16.0 updates form previous pull)
-Fixed and replaced all screenshots
-Removed use of 2nd person
-Reworked example flow in doc

Cleaned up RST, including:
-Fixed 100th character line breaks
-Fixed indentations
-Fixed typos and improved grammar

### CODE REVIEW

**Comment by brse-odoo** [comment]
Hi @ksc-odoo -- when you have a chance, can you please review this doc? Thanks in advance for your help on this!

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:7]: A *Bill of Materials* (or *BoM*) is a document that defines the quantity of each component required
to manufacture (or deliver) a finished product. It can also include various operations types, 
steps, and instructions for the individual guidelines needed to complete a production process.

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:10]: In Odoo Manufacturing, multiple :abbr:`BoMs (Bills of Materials)` can be linked to a product, so
even product variants can have their own specific :abbr:`BoMs (Bills of Materials)`.

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:12]: Correctly setting up a :abbr:`BoM (Bill of Materials)` helps optimize the manufacturing process, and saves time.

Would this use of ":abbr:" work in this instance? Not quite sure - totally your call.

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:20]: :abbr:`BoMs (Bills of Materials)` can be set up with - or without - operations steps or instructions.
They can include as many (or as few) components, as needed. The simplest setup is one without
operations (or instructions). In that case, the production is solely managed with *Manufacturing
Orders*.


Would this final phrase require a :guilabel: formatting?

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:24]: To create a :abbr:`BoM (Bill of Materials)`, navigate to :menuselection:`Manufacturing -->
Products --> Bills of Materials`, and click :guilabel:`Create`. Then, in the :guilabel:`Product`
field, choose the product that will be associated with the :abbr:`BoM (Bill of Materials)`.

**Review by tiku-odoo — COMMENTED** [review]
Great work on this @brse-odoo. I have noted a few suggestions in addition to the changes that KC requested. Last section-- the tab was labeled incorrectly. Can you confirm I changed it to the correct tab: By-Products? 

Also after taking a look at the original doc, does this note still hold as true?:  

Note

Each operation is unique, as it is always exclusively linked to one BoM. Operations can be reused when configuring a new BoM, with the Copy Existing Operations feature.

Might be good to include. 

Looking forward to seeing this published, nice work. 

Thanks,
Tim

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:12]: Yes, it would work in this instance and help keep things consistent from the start. Good catch!

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:20]: Good changes for the rest of the paragraph, KC! But Tim's right here - no `:guilabel:` is needed since we're not referring to UI on screen.

**Comment by brse-odoo** [comment]
Hi @tiku-odoo, thanks for reviewing! Nice catch at the end -- you did change it to the correct tab `By-Products`. For the `..note::`, though, I removed that for this 14.0-specific version since it doesn't apply until 15.0. Thank you!

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @brse-odoo 👋 I've made some technical changes on this doc for your reference, overall this doc looks really good! Nice job 🙂.

Most of my changes were on :guilabels:, italics, and indenting. I will push these changes in a second commit after this review for your approval. Once approved I will squash the commits, rebase, and push back up for final review. Thank you!

**Comment by brse-odoo** [comment]
> Hi @brse-odoo 👋 I've made some technical changes on this doc for your reference, overall this doc looks really good! Nice job 🙂.
> 
> Most of my changes were on :guilabels:, italics, and indenting. I will push these changes in a second commit after this review for your approval. Once approved I will squash the commits, rebase, and push back up for final review. Thank you!

Hi @samueljlieber, thanks for completing the technical review and implementing these changes! Everything looks good to me and you have my approval on the changes.

**Review by samueljlieber — APPROVED** [review]

**Comment by brse-odoo** [comment]
Hi @StraubCreative, this PR is ready for your final review whenever you have a chance!

**Review by StraubCreative — CHANGES_REQUESTED** [review]
Hi @brse-odoo 
Confirm we're using sentence-case headings and that 100th character line breaks are correct, and tag me again for review.
Thanks

**Comment by brse-odoo** [comment]
@StraubCreative The 100th character line breaks have been fixed. This is ready for final review

**Review by StraubCreative — CHANGES_REQUESTED** [review]
I'm still seeing 100th character issues here.
cc: @samueljlieber

**Comment by brse-odoo** [comment]
@samueljlieber If you have the time this could use one last technical look-over before heading to final review. I just edited and pushed up revisions. Thank you!

File: content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst
Comment (StraubCreative) [inline content/applications/inventory_and_mrp/manufacturing/management/bill_configuration.rst:72]: Ok so now we know there are two ways to assign BoMs to product variants, so... where's the walk-through? 😅 

Also, I think this a good spot to add a **seealso** for [this eLearning walk-through](https://www.odoo.com/slides/slide/manufacturing-with-product-variants-2805). Jose does a pretty good job showing exactly how to do this.

**Review by StraubCreative — APPROVED** [review]
Hi @brse-odoo 

Approving this PR, with comments.

We can publish it as it is, however I have a few suggestions that I think would really help it's readability and comprehension in a couple spots. There are 3 suggested changes in total so it's not a big lift.

From here, you can choose to merge right away or can address the changes and then merge.

**Important:** In either case, since this isn't a forward-port, we will need to issue the merge commands as follows, with one command per line.
```
[at]robodoo r+
[at]fw-bot ignore
```

Please let me know if you have any questions or need help in proceeding from here.

**Pull request closed without merging**
